### PR TITLE
YDBOPS-9679 dynconfig CMS ReplaceConfig

### PR DESCRIPTION
**Pull request by kobzonega**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: YDBOPS-9679

## What is the new behavior?

- validate dynconfig for kubernetes (must contain [mandatory blobstorage fields](https://github.com/ydb-platform/ydb-kubernetes-operator/pull/214/files#diff-3059c6689a931622298207f1a7299cc00d66d9df4f459f4f8372605b097dbf69R149))
- insert `.config` field without changes from dynconfig into ConfigMap as static config
- check the need for ReplaceConfig by compare metadata.version with CMS GetConfig request
- update config in cluster by send dynconfig with CMS ReplaceConfig request

### CODE REVIEW

File: internal/cms/dynconfig.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		storage.GetStorageEndpointWithProto(),
+		storage.Spec.Domain,
+	)
+	conn, err := connection.Open(ctx,
+		endpoint,
+		ydb.WithCredentials(creds),
+		ydb.MergeOptions(opts...),
+	)
+	if err != nil {
+		logger.Error(err, "Error connecting to YDB storage")
```
Comment (nmerkulov) [inline internal/cms/dynconfig.go:44]: up to you насчет исправления

вместо логгирования одной и той же ошибки два раза стоит добавить контекст и вернуть ее наверх

т.е.

return nil, fmt.Errorf("Failed to connect to YDB: %w", err)

File: internal/cms/dynconfig.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
```
Comment (nmerkulov) [inline internal/cms/dynconfig.go:93]: то же самое, что и с другим коннектом

File: internal/cms/operation.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		storage.GetStorageEndpointWithProto(),
+		storage.Spec.Domain,
+	)
+	conn, err := connection.Open(ctx,
+		endpoint,
+		ydb.WithCredentials(creds),
+		ydb.MergeOptions(opts...),
+	)
+	if err != nil {
+		logger.Error(err, "Error connecting to YDB storage")
```
Comment (nmerkulov) [inline internal/cms/operation.go:37]: same

**Comment by nmerkulov** [comment]
few nits

overall lgtm

**Review by artgromov — APPROVED** [review]

**Pull request merged**